src/utils/parsers.ts

```diff
@@ -208,6 +208,8 @@ function parseEnvArgs(
     [],
     [],
   ];
+  console.error('val', value)
+  console.error('cur', current)
   if (current[1].length === 0) {
     // Parse a secret path
     if (value !== '--') {
```

Comment:
I have tried to think up of many solutions, but I can't find a way to consistently use `'--'` as a delimiter between the secrets and a command using commander. The simplest way to fix this would be to use commander to process and validate the arguments, and use `process.env` to get the raw arguments, which will include the `'--'`. We can parse the raw arguments and use them reliably, but I don't like the fact that we are basically bypassing the literal purpose of commander just to get something like this working.

This is kind of broken right now due to the option for removing newlines. If there are zero arguments, but the command doesn't fail, that means that there were some paths provided with the option to preserve the newlines. However, in this case, the `this.passThroughOptions()` doesn't pass through `'--'`, as it only does that after we have started listing the arguments, which didn't happen in this case, as we are listing individual paths inside the preserve newline option, so commander gobbles up the `'--'`. This can be seen in this command:

```console
[aryanj@matrix-34xx:~]$ polykey secrets env --preserve-newline vault:SECRET -- command
# args = [['command', undefined, undefined], []]
# the -- gets parsed by commander instead of the parser
```

We need to address this problem in this PR, but I'm not sure how I can even approach this. I need inputs.

Reply:
Related issue: https://github.com/MatrixAI/Polykey-CLI/issues/325

Reply:
Answered in https://github.com/MatrixAI/Polykey-CLI/pull/329#issuecomment-2472332972